### PR TITLE
PHP Version constraint: stick to tested versions only

### PR DESCRIPTION
**Pull request by Slamdunk**
PHP doesn't follow SemVer, so we should provide support only for tested PHP versions

### CODE REVIEW

**Review by Ocramius — CHANGES_REQUESTED** [review]
LGTM, **but** we can add PHP 8.2 at this stage, since it is stable-ish.

Remember to:

 1. [x] define `config.platform.php` to help dependabot/etc in upgrades
 2. [x] update `composer.lock` to reflect the current `composer.json`

**Comment by Slamdunk** [comment]
> 1. define `config.platform.php` to help dependabot/etc in upgrades

Can you elaborate a bit please? Can't find any doc about it :thinking:

File: composer.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
         }
     ],
     "require": {
-        "php": "~8.1.0",
+        "php": "~8.1.0 || ~8.2.0",
```
Comment (Ocramius) [inline composer.json:20]: 8.2 should also be in CI before merge

Comment (Slamdunk) [reply composer.json:20]: https://github.com/lcobucci/jwt/blob/2fecef3210030c1bfaead33116228118b7ca9cc0/.github/workflows/phpunit.yml#L77

**Comment by Ocramius** [comment]
> > 1. define `config.platform.php` to help dependabot/etc in upgrades
> 
> Can you elaborate a bit please? Can't find any doc about it thinking

If we declare `{"require": {"php": "~8.1.0 || ~8.2.0"}}`, then @dependabot will pick up packages compatible with PHP 8.2.0 **only** (at some point).

To work around this, we can declare `{"config": {"platform": {"php": "8.1.99"}}}` to force Composer's SAT logic to always pick a dependency set that is compatible with PHP 8.1.x, while running `composer update`.

**Comment by Slamdunk** [comment]
> `{"config": {"platform": {"php": "8.1.99"}}}`

Ah, LoL, yes, I was looking for a `config.platform.php` file :facepalm:

**Review by Ocramius — APPROVED** [review]
LGTM, letting @lcobucci call the shots though

**Review by lcobucci — APPROVED** [review]
Ship it!

**Pull request merged**